src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -210,26 +272,42 @@ private void handleMigrationOfOldData()
 		if (Strings.isNullOrEmpty(hasMigratedToV2))
 		{
 			log.info("Migrating data to V2");
-			// Perform migration of old data
-			Type setupType = new TypeToken<ArrayList<InventorySetup>>()
+			inventorySetups.addAll(loadV1Setups());
+			updateConfig(true, false);
+			inventorySetups.clear();
+			configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2, "True");
+		}
+
+		hasMigratedToV2 = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2);
+		if (!Strings.isNullOrEmpty(hasMigratedToV2))
+		{
+			String oldData = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS);
+			if (oldData != null)
 			{
+				log.info("Removing old v1 data key");
+				configManager.unsetConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS);
+			}
+		}
 
-			}.getType();
-			inventorySetups.addAll(loadData(CONFIG_KEY_SETUPS, setupType));
+		String hasMigratedToV3 = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V3);
+		if (Strings.isNullOrEmpty(hasMigratedToV3))
+		{
+			log.info("Migrating data to V3");
+			inventorySetups.addAll(loadV2Setups());
 			updateConfig(true, false);
 			inventorySetups.clear();
-			configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2, "True");
+			configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V3, "True");
 		}
 
 		// TODO Don't unset configuration until new version is stable
-//		hasMigratedToV2 = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2);
-//		if (!Strings.isNullOrEmpty(hasMigratedToV2))
+//		hasMigratedToV3 = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V3);

```

Comment:
you can remove the assignmetn and wrapping if statement here too, (but still leave it commneted out

Reply:
done